### PR TITLE
Feature/21 builtin users

### LINKED ISSUE #21: Built-in user list in config
## Background
I can imagine that for future development, we will need to run the login-gateway without any real connection, just for testing purposes but not testing of the login-gateway.

I propose we have the ability to run the login-gateway and pass the JSON config of the users to it rather than a config for AD or other services.

## Feature
Add a run for integration tests options

## Example of config JSON
```json
[
  {
    "user": "user1",
    "password": "password1"
    "roles": ["user", "team1", "view"] 
  },
  {
    "user": "admin1",
    "password": "adminpassword1"
    "roles": ["admin", "wheel"] 
  },
]
```

### PR DESCRIPTION
**Pull request by dk1844**
This PR introduces configuration of the user section where known users can be added (instead of previously built-in) single user.

Configuration redone to YAML in order to properly set list of user details.
Individual configuration classes split - they feature their own validations -- that are run on startup (preventing the application to run if invalid).

Tests added for config classes and the related auth provider.

Closes #21

### CODE REVIEW

File: service/src/main/scala/za/co/absa/logingw/rest/config/ConfigValidatable.scala [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.logingw.rest.config
+
+trait ConfigValidatable {
+
+  /** May throw ConfigValidationException */
+  def validate(): Unit
```
Comment (jakipatryk) [inline service/src/main/scala/za/co/absa/logingw/rest/config/ConfigValidatable.scala:22]: Have you thought about using `Try[Unit]` at this level? Or a custom trait `ConfigValidationResult` and case object `ConfigValidationSuccess` and case class `ConfigValidationError(reason)`

**Review by jakipatryk — COMMENTED** [review]
LGTM (just checked the code for now), will approve once the problem caused by my PR is fixed (sorry)

File: service/src/main/scala/za/co/absa/logingw/rest/SecurityConfig.scala [hunk 2]
```diff
@@ -55,26 +56,27 @@ class SecurityConfig {
     http.build()
   }
 
+  @Autowired
+  val usersConfig: UsersConfig = null
```
[the remark this answers is not in the document]
Comment (dk1844) [reply service/src/main/scala/za/co/absa/logingw/rest/SecurityConfig.scala]: Yes, there were problems with wirings, so I made another component object - it already has constructor autowiring for these fields.

File: service/src/main/scala/za/co/absa/logingw/rest/SecurityConfig.scala [hunk 3]
```diff
@@ -55,26 +56,27 @@ class SecurityConfig {
...
+
+  // TODO make it autowired but only if in config AD LDAP provider is set up (#28)
+  @Autowired
+  val adLDAPConfig: ActiveDirectoryLDAPConfig = null
```
[the remark this answers is not in the document]
Comment (dk1844) [reply service/src/main/scala/za/co/absa/logingw/rest/SecurityConfig.scala]: ditto

File: service/src/main/resources/application.yaml
[the remark this answers is not in the document]
Comment (dk1844) [reply service/src/main/resources/application.yaml:1]: Agreed, unified all strings to use `""` and integer to be just numbers.

File: service/src/main/resources/application.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+logingw:
+  rest:
+    jwt:
+      exp-time: '4'
```
[the remark this answers is not in the document]
Comment (dk1844) [reply service/src/main/resources/application.yaml]: Agreed, changed.

File: service/src/main/scala/za/co/absa/logingw/rest/config/UsersConfig.scala [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  def init() {
+    this.validate()
+  }
+}
+
+@ConstructorBinding
+case class UserConfig(
+  username: String,
+  password: String,
+  email: String, // may be null
```
[the remark this answers is not in the document]
Comment (dk1844) [reply service/src/main/scala/za/co/absa/logingw/rest/config/UsersConfig.scala:58]: Sadly, it does not. So we can wrap it with Option when used or create another method to pre-wrap it.

File: service/src/main/scala/za/co/absa/logingw/rest/config/ConfigValidatable.scala (hunk 1 above)
Comment (dk1844) [reply service/src/main/scala/za/co/absa/logingw/rest/config/ConfigValidatable.scala:22]: Thanks for the idea, I followed it.
Originally, I just wanted the config validation to fail on an invalid config and that was it. It still does, but now there is a wrapper (_at the end of the world_ sort of), but the validation is done more idiomatically.

**Review by jakipatryk — APPROVED** [review]
LGTM (checked code, ran locally)

File: service/src/main/scala/za/co/absa/logingw/rest/config/ConfigValidatable.scala (hunk 1 above)
Comment (jakipatryk) [reply service/src/main/scala/za/co/absa/logingw/rest/config/ConfigValidatable.scala]: Looks cool!

File: service/src/main/resources/application.yaml [hunk 6]
```diff
@@ -19,16 +21,40 @@ logingw:
...
 server:
   port: 9090
+
+# Health Check Config + JMX Config
+springdoc:
+  show-actuator: true
+
+management:
+  health:
+    #TODO: Enable Ldap check when fully Implemented - issue #20
```
[the remark this answers is not in the document]
Comment (dk1844) [reply service/src/main/resources/application.yaml]: This is about LDAP health check not working at the moment (`LdapHealthIndicator` is by default autoconfigured to be part of the actuator health check), but we are missing something.

Perhaps poor wording, originally, redone.
Better described in issue #34

File: service/src/main/resources/application.yaml [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+          password: "password1"
+          groups:
+        -
+          username: "TestUser"
+          password: "password123"
+          email: "test@abs.com"
+          groups:
+            - "groupA"
+            - "groupB"
+            -
```
Comment (jakipatryk) [inline service/src/main/resources/application.yaml:24]: I think this is unintended - with empty `-` I am getting an error

Comment (dk1844) [reply service/src/main/resources/application.yaml]: Thank you! Yes, it was unintended (newline in editor auto-created another `-`-prefixed field

**Review by jakipatryk — APPROVED** [review]
LGTM

**Pull request merged**